### PR TITLE
Working Joe Death Sound and Death Message (It is so Joever)

### PR DESCRIPTION
**Pull request by Ben10083**
# About the pull request

This PR gives Working joes a unique death sound and death message....upon death.

TODO:
- [x] Add framework for implementation.
- [x] Find appropriate sounds for Working Joe death
- [x] (Optional) Have a random death noise be picked to be played
- [x] Implement a death message for Working Joes

# Explain why it's good for the game

For the same reasons we added the initial Working Joe audible emotes, this will add more depth to the Joes and make them even more uncanny. (Also default death message makes no sense as Working Joe eyes are already dead and lifeless.)

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/cmss13-devs/cmss13/assets/91219575/f266f4f3-983c-4f2e-907f-97218eb939f7

Example of default death noise playing, what most will hear.

</details>

# Changelog

:cl:
add: Working Joes now have a unique death message. Credit to Quickload for the message.
soundadd: Working Joes now have a death rattle. Credit to Quickload for shifting through Alien Isolation audio files.
qol: Things that die with intent eyes now lose color in their eyes on death.
/:cl:

### CODE REVIEW

**Comment by Ben10083** [comment]
This should be good enough to put up for review, other sounds for death occur at about 5% chance

File: code/modules/mob/living/carbon/human/species/synthetic.dm [hunk 1]
```diff
@@ -109,13 +109,19 @@
...
+	death_message = "violently gargles fluid and seizes up, the glow in their eyes dimming..."
 	uses_ethnicity = FALSE
 	mob_inherent_traits = list(TRAIT_SUPER_STRONG, TRAIT_INTENT_EYES, TRAIT_EMOTE_CD_EXEMPT)
 
 	hair_color = "#000000"
 	icobase = 'icons/mob/humans/species/r_synthetic.dmi'
 	deform = 'icons/mob/humans/species/r_synthetic.dmi'
 
+// Special death noise for Working Joe
+/datum/species/synthetic/colonial/working_joe/handle_death(mob/living/carbon/human/H, gibbed)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/species/synthetic.dm:121]: Call parent

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species/synthetic.dm (hunk 1 above)
Comment (Ben10083) [reply code/modules/mob/living/carbon/human/species/synthetic.dm:121]: Will do so, but don't really see any synthetic handle_death to call

File: code/modules/mob/living/carbon/human/species/working_joe/_species.dm [hunk 2]
```diff
@@ -14,6 +15,11 @@
 	. = ..()
 	give_action(joe, /datum/action/joe_emote_panel)
 
+// Special death noise for Working Joe
+/datum/species/synthetic/colonial/working_joe/handle_death(mob/living/carbon/human/H, gibbed)
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/species/working_joe/_species.dm:19]:
```suggestion
/datum/species/synthetic/colonial/working_joe/handle_death(mob/living/carbon/human/dying_human, gibbed)
```

File: code/modules/mob/living/carbon/human/species/working_joe/_species.dm [hunk 3]
```diff
@@ -14,6 +15,11 @@
...
+	if(!gibbed) //A gibbed Joe won't have a death rattle
+		playsound(H.loc, pick_weight(list('sound/voice/joe/death_normal.ogg' = 75, 'sound/voice/joe/death_silence.ogg' = 10, 'sound/voice/joe/death_tomorrow.ogg' = 10,'sound/voice/joe/death_dream.ogg' = 5)), 25, FALSE)
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/species/working_joe/_species.dm:21]:
```suggestion
		playsound(dying_human.loc, pick_weight(list('sound/voice/joe/death_normal.ogg' = 75, 'sound/voice/joe/death_silence.ogg' = 10, 'sound/voice/joe/death_tomorrow.ogg' = 10,'sound/voice/joe/death_dream.ogg' = 5)), 25, FALSE)
```

File: code/modules/mob/living/carbon/human/species/working_joe/_species.dm [hunk 4]
```diff
@@ -14,6 +15,11 @@
...
+	. = ..()
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/species/working_joe/_species.dm:22]:
```suggestion
	return ..()
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**